pkg/sessions/redis/redis_store.go

```diff
@@ -114,20 +126,33 @@ func (store *SessionStore) SaveUserSession(ctx context.Context, s *sessions.Sess
 	} else {
 		ticket = string(keyVal)
 	}
-	err := s.ObtainLock(ctx, exp)
-	if err != nil {
-		return fmt.Errorf("error occurred while trying to obtain lock: %v", err)
-	}
-	defer func() {
-		if s == nil {
-			return
+	return saveTicketWithLock(ctx, store, keyName, ticket, exp)
```

Comment:
So... this function is generally speaking doing a "read-modify-write" of the document stored under keyName. As the code currently stands, it doesn't get the lock until it is about to do the write. Instead, the lock needs to be obtained before the read (to be useful in avoiding races).

Reply:
thanks @joel-bluedata . done the change